src/Aspire.Hosting.Dapr/DaprDistributedApplicationLifecycleHook.cs

```diff
@@ -192,6 +199,10 @@ public async Task BeforeStartAsync(DistributedApplicationModel appModel, Cancell
                         {
                             updatedArgs.AddRange(daprProfilePortArg($"{{{{- portForServing \"{daprSideCarResourceName}_profile\" -}}}}")());
                         }
+                        if (sidecarOptions?.AppChannelAddress is null)
+                        {
+                            updatedArgs.AddRange(daprAppChannelAddressArg($"{{{{- addressForServing \"{daprSideCarResourceName}_grpc\" -}}}}")());
```

Comment:
This should be the address for the application (i.e. service), not the sidecar, right?

Reply:
Technically, yes, but the model that we have for Executables in Aspire goes beyond what DAPR sidecar currently supports. Namely, an Executable can expose multiple services and each of those can bind to a different address. So I am making a simplifying assumption here that there will be just one address that both the Executable and the DAPR sidecar use to bind everything to it. It does not have to be `127.0.0.1` or `[::1]`, but it is shared between the sidecar and all services that an Executable exposes.

Happy to put a comment about this/open to other suggestions.

Reply:
Had a discussion with @philliphoff offline. Updated the code so that it uses the http endpoint, if it exists, to figure out the address that the DAPR sidecar needs to use to talk to the service process